subprojects/docs/src/samples/maven/pomGeneration/kotlin/build.gradle.kts

```diff
@@ -0,0 +1,117 @@
+// tag::use-plugin[]
+plugins {
+    // end::use-plugin[]
+    war
+// tag::use-plugin[]
+    maven
+}
+// end::use-plugin[]
+
+group = "gradle"
+version = "1.0"
+base.archivesBaseName = "mywar"
+buildDir = file("target")
+
+repositories {
+    flatDir {
+        dirs("lib")
+    }
+}
+
+configurations.runtime.exclude(group = "excludeGroup2", module = "excludeArtifact2")
+
+dependencies {
+    compile("group1:compile:1.0") {
+        exclude(group = "excludeGroup", module = "excludeArtifact")
+    }
+    // NOTE: explicit artifact reference makes a dependency non-transitive
+    providedCompile("group2:providedCompile:1.0@jar")
+    runtime("group3:runtime:1.0")
+    providedRuntime("group4:providedRuntime:1.0@zip") {
+        artifact {
+            name = "providedRuntime-util"
+            type = "war"
+        }
+    }
+    testCompile("group5:testCompile:1.0")
+    testRuntime("group6:testRuntime:1.0")
+}
+
+// Include a javadoc zip
+
+task<Zip>("javadocZip") {
+    dependsOn("javadoc")
+    classifier = "javadoc"
+    from(tasks.getByName<Javadoc>("javadoc").destinationDir)
+}
+
+artifacts {
+    add("archives", tasks["javadocZip"])
+}
+
+// Configure the release and snapshot repositories
+
+tasks.getByName<Upload>("uploadArchives") {
+    repositories.withGroovyBuilder {
+        "mavenDeployer" {
+            "repository"("url" to uri("pomRepo"))
+            "snapshotRepository"("url" to uri("snapshotRepo"))
+        }
+    }
+}
+
+// Customize the contents of the pom
+
+// tag::when-configured[]
+val install by tasks.getting(Upload::class)
+val uploadArchives by tasks.getting(Upload::class)
+val installer = install.repositories.withGroovyBuilder { getProperty("mavenInstaller") as MavenResolver }
+val deployer = uploadArchives.repositories.withGroovyBuilder { getProperty("mavenDeployer") as MavenResolver }
+
+listOf(installer, deployer).forEach {
+    it.pom.whenConfigured {
+        dependencies.firstOrNull { dep ->
+            dep?.withGroovyBuilder {
```

Comment:
:x: Can `dep` really be null here? Otherwise I'd write `dep!!` and get rid of the default.